lib/logstash/filters/geoip.rb

```diff
@@ -169,16 +169,18 @@ def filter(event)
       raise e
     end
 
-    event.set(@target, geo_data_hash)
+    success = apply_geodata(geo_data_hash, event)
 
     if geo_data_hash.empty?
       tag_unsuccessful_lookup(event)
       return
     end
 
-    filter_matched(event)
+    if success
```

Comment:
should we take care of the else here? if apply_geodata "fails", then there is no tag but also no filter_matched